components/canvas/canvas_paint_task.rs

```diff
@@ -255,6 +226,7 @@ impl<'a> CanvasPaintTask<'a> {
                     CanvasMsg::PutImageData(imagedata, image_data_rect, dirty_rect)
                         => painter.put_image_data(imagedata, image_data_rect, dirty_rect),
                     CanvasMsg::Close => break,
+                    _ => panic!("Wrong message sent to Canvas2D task"),
```

Comment:
This is why multiple enums are better: this catchall will suppress nonexhaustive-match compiler warnings.